src/snowflake/snowpark/mock/plan.py

```diff
@@ -1338,11 +1360,15 @@ def calculate_expression(
                         )
                     else:
                         res_cols.append(target_expr)
-
             res_col = ColumnEmulator(
                 data=res_cols, dtype=object
             )  # dtype=object prevents implicit converting None to Nan
             res_col.index = res_index
+            res_col.sf_type = (
+                calculated_sf_type
+                if calculated_sf_type
+                else ColumnType(NullType(), True)
```

Comment:
We should not use `NullType` here.

Reply:
I keep the NullType here just for defensive coding for the case if all the windows are return None